src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse.PlatformNotSupported.cs

```diff
@@ -39,6 +39,11 @@ public static class Sse
         /// </summary>
         public static Vector128<float> AndNot(Vector128<float> left, Vector128<float> right) { throw new PlatformNotSupportedException(); }
 
+        /// <summary>
+        /// HELPER: MOVSS
+        /// </summary>
+        public static Vector128<float> CastSingleToVector128Single(float value) { throw new PlatformNotSupportedException(); }
```

Comment:
The opposite operation seems to be `ConvertToSingle` so it's not clear if "Cast" is a good choice of name for this one. Oh well, even Convert is not a great name IMO as it's really a form of extract.

Reply:
CC. @eerhardt, @CarolEidt.

We should decide on a consistent naming convention for these type of intrinsics.

Reply:
We already have 

`Avx.BroadcastScalarToVector128`
`Sse2.LoadScalarVector128`

that both take pointers.

I agree `Cast` doesn't seem like a great name here.  What's the difference between this and `LoadScalar`?

Reply:
According to @fiigii (https://github.com/dotnet/coreclr/pull/16772#issuecomment-370713432) these are meant to behave as `StaticCast` in that they do not normally generate code (at least not when the source and destination register are the same).

@fiigii:
* For a value already in the target register, this will produce no codegen.
* For a value going to a different target register, will this generate `movaps`?
* What is the behavior when `value` needs to be loaded from memory?
  * Will it just do `movss` and upper will be "indeterminate" or will it behave as `SetScalar` and ensure upper is clear?

Reply:
IMO naming convention could be based on `StaticCast`. We could add additional overloads for `float` and `double` and perhaps for all supported integral types as well. Unfortunately on x86 we would need jit support for 8 byte integrals best in the form where double operations would be used.

Reply:
> Will it just do movss and upper will be "indeterminate" or will it behave as SetScalar and ensure upper is clear?

Yes, they will try to reuse the underlying representation of float/double as Vector128, so that the movss/movsd will be generated by variable access and upper will be arbitrary content.

Question. Shall we add the `unsafe` modifier for these two intrinsics because of the arbitrary upper values?

Reply:
@CarolEidt, does this sound like okay behavior to you?

It seems "bad" to me that we would allow non-deterministic behavior here (and that it would only be non-deterministic in one case, which is where `float` is loaded from memory and is not already in a register).

Reply:
> The case where it is already in the desired register should have the same behavior as the case where it needs to be moved.

Right, and the current behavior @fiigii is suggesting will work for those two scenarios.

If it is already in the target register, it is a no-op
If it is already in register, but not the target, it will generate a `movaps` and will copy all bits

However, the current suggested behavior would be non-deterministic in the case where `value` is not already in register and needs to be read from memory.

Currently, @fiigii is suggesting that we read using `movss` (as required) but leave the upper bits indeterminate.

I believe this could be problematic and would rather see the upper bits zero'd (equivalent to calling `SetScalarVector128`).

That being said, even zero'ing the bits may lead to non-deterministic behavior as the value of the upper bits will depend on things like how the JIT allocates registers and whether the value gets spilled to the stack or not.

Reply:
@CarolEidt @tannergooding Fair enough. I will remove them and try to optimize SetScalarVector128 later.